src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -74,6 +74,24 @@
             _memoryCache.Remove(cacheKey);
             _logger.LogDebug($"[IdWebCache] MemoryCache: Remove cacheKey {cacheKey} Time in Ticks: {Utility.Watch.Elapsed.Ticks - startTicks}. ");
 
+            if (!_distributedCacheOptions.AwaitL2CacheOperation)
+            {
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+                Task.Run(async () =>
+                {
+                    await DistributedCacheRemoveKeyAsync(cacheKey, startTicks).ConfigureAwait(false);
+                });
+#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+                await Task.CompletedTask.ConfigureAwait(false);
+            }
+            else
+            {
+                await DistributedCacheRemoveKeyAsync(cacheKey, startTicks).ConfigureAwait(false);
```

Comment:
>await DistributedCacheRemoveKeyAsync(cacheKey, startTicks).ConfigureAwait(false); [](start = 16, length = 81)

should be an action, and the work should be added to an queue, and there should be an event processor to guarantee ordering of events. should do a bit more work here